File: frontend/src/app/api-generated/models/booking-place-with-bookings-view-model.ts
/* tslint:disable */
/* eslint-disable */
import { BookingPlaceItemTypeEnum } from '../models/booking-place-item-type-enum';
import { BookingPlaceTypeEnum } from '../models/booking-place-type-enum';
import { BookingViewModel } from '../models/booking-view-model';
import { UserViewModel } from '../models/user-view-model';
export interface BookingPlaceWithBookingsViewModel {
  bookings?: Array<BookingViewModel> | null;
  floorId?: number;
  id?: number;
  itemType?: BookingPlaceItemTypeEnum;
  name?: string | null;
  reservedForUserId?: number | null;
  reservedForUserVm?: UserViewModel;
  type?: BookingPlaceTypeEnum;
}

File: backend/ViewModels/BookingPlaceWithBookingsViewModel.cs
using BookingApp.Enums;

namespace BookingApp.ViewModels
{
    public class BookingPlaceWithBookingsViewModel
    {
        public int Id { get; set; }
        public required string Name { get; set; }
        public BookingPlaceTypeEnum Type { get; set; }
        public BookingPlaceItemTypeEnum ItemType { get; set; }
        public bool AvailableForBooking { get; set; }
        public DateTime? AvailableFrom { get; set; }
        public DateTime? AvailableTo { get; set; }
        public int FloorId { get; set; }
        public int? ReservedForUserId { get; set; }
        public UserViewModel? ReservedForUserVm { get; set; }
        public List<BookingViewModel> Bookings { get; set; } = new List<BookingViewModel>();
    }
}
